### PR TITLE
Add helper package for looking up PCI/USB device & vendor names

### PR DESCRIPTION
**Pull request by jpnurmi**
Yet another enabler for #77, for looking up human-readable network device models and vendor names that are not provided by nm.

```
ubuntu-desktop-installer/packages/udev/example $ dart run main.dart 
Wired devices:
	Linksys Gigabit Ethernet Adapter
	Realtek Semiconductor Corp. RTL8153 Gigabit Ethernet Adapter
Wireless devices:
	Qualcomm Atheros QCA6174 802.11ac Wireless Network Adapter
	D-Link Corp. DWA-131 Wireless N Nano Adapter (Rev. E1) [Realtek RTL8192EU]
	Ralink Technology, Corp. RT5370 Wireless Adapter
```

### CODE REVIEW

**Comment by mwhudson** [comment]
I know we have by and large decided that the desktop installer will not use subiquity's code for configuring the network (as subiquity assumes netplan) but it might make sense to use it to discover available links, at least. Because it already has code for poking around in sysfs and udev and so on for model / vendor etc and one implementation of all that garbage is probably enough? (One feature it has that this doesn't seem to is to save the network state on one system and use it to run the UI on another)

**Comment by mwhudson** [comment]
Oh and this seems to be using your own library to map pci ids to model/vendor? We already have one of those in udev...

**Comment by jpnurmi** [comment]
> We already have one of those in udev...

Cool, thanks! I had checked what lspci does... :smile:

**Comment by jpnurmi** [comment]
I have added absolute minimal libudev FFI bindings with only the few methods that are needed for #77. It might make sense to create a separate complete package later.

**Comment by jpnurmi** [comment]
@oSoMoN, who would be the right person to review this?

**Comment by jpnurmi** [comment]
If someone wants to try it out, there's a tiny example (`dart run example/example.dart`). It's not necessary to grab the entire Connect to Internet page unless you want to. 🙂

**Comment by seb128** [comment]
@robert-ancell could you maybe help reviewing this one?

**Review by robert-ancell — APPROVED** [review]
LGTM. This seems like it would be a generally useful package for other projects too.

File: packages/udev/lib/udev.dart [hunk 1]
```diff
@@ -0,0 +1,31 @@
+/// {@template udev}
+/// Provides APIs to introspect and enumerate devices on the local system.
+///
+/// ```dart
+/// void main() async {
+///   final client = NetworkManagerClient();
```
[the remark this answers is not in the document]
Comment (jpnurmi) [reply packages/udev/lib/udev.dart:6]: There's a separate `pubspec.yaml` in the example to make it fetch the dependency. It works if you run `dart pub get` and `dart run example.dart` in the `example/` directory. I wanted to illustrate how the package is used for fetching device model and vendor names in the installer. :)

I'd kill the dependency too if we made this a separate standalone and complete package. For now, this is an absolutely minimal internal package for the installer. There's not even API for enumerating devices, only the very few members needed for the "connect to internet" page.

**Pull request merged**